packages/date-time-picker/src/vaadin-date-time-picker.js

```diff
@@ -634,11 +634,11 @@ class DateTimePicker extends FieldMixin(
 
   /** @private */
   __i18nChanged(changeRecord) {
-    if (this.__datePicker) {
+    if (this.__datePicker && this.__datePicker.$) {
```

Comment:
We found that the following test fails:

```js
const TemplateClass = templatize(html`<div>
  <vaadin-date-time-picker></vaadin-date-time-picker>
</div>`);

const instance = new TemplateClass();
const wrapper = instance.root.querySelector('div');
const dateTimePicker = wrapper.querySelector('vaadin-date-time-picker');
dateTimePicker.i18n = { ...dateTimePicker.i18n, cancel: 'CUSTOM' };
document.body.appendChild(wrapper);
const datePicker = document.createElement('vaadin-date-time-picker-date-picker');
expect(datePicker.i18n.cancel).to.equal('CUSTOM');
```

...but it fails regardless of the workaround actually.

Reply:
In the above example using templatize, Polymer invokes `ready()` for date-time-picker even if you don't add the wrapper to the DOM. This is very weird especially given that connectedCallback and disconnectedCallback are not invoked.

Reply:
We found the proper solution. This thread is not relevant anymore.